### ISSUE TITLE
Unhandled exception `Bad state: No element` with `CliIntegration` plugin sending a Ctrl+C to bot

### ISSUE THREAD

**Issue by V3ntus** [nyxx-discord/nyxx#450]
**Describe the bug**
Bot hangs when sending SIGINT/Ctrl+C while calling `onBotStop` for plugins. Seems to happen on `StreamSubscription` cancellation with any of nyxx's plugins.

**To Reproduce**
1. Start bot with `Logging` and `CliIntegration` plugins.
2. Ctrl+C to trigger dispose

**Expected behavior**
Proper disposal.

**Screenshots**
```
^C[2023-03-17 18:49:07.872802] [INFO] [Client] Disposing and closing client...
[2023-03-17 18:49:07.873071] [FINE] [Client] Triggering onBotStop for plugin: Logging
Unhandled exception:
Bad state: No element
#0      Stream.firstWhere.<anonymous closure> (dart:async/stream.dart:1709:9)
#1      _RootZone.runGuarded (dart:async/zone.dart:1581:10)
#2      _DoneStreamSubscription._sendDone (dart:async/stream_impl.dart:699:36)
#3      _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#4      _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#5      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:123:13)
#6      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:405:11)
#7      _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#8      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192:26)
```

**Desktop (please complete the following information):**
 - OS: Mac OS x64 Intel
 - Dart version: 2.19.4
 - Nyxx version: 5.0.0

**Comment by abitofevrything** [comment]
Hi, I don't know where you're getting this log line from: `Triggering onBotStop for plugin: Logging`. Are you running a custom fork of nyxx?

I also can't reproduce this on our `main` branch. Could you send the code you're using to reproduce this?

**Comment by abitofevrything** [comment]
I've been able to reproduce and have submitted a fix in #451. For now, you can just press ctrl-c again to forcibly terminate the process before the error occurs.

**Comment by V3ntus** [comment]
Hello! Yes this is a fork, although these log lines should be the only thing added that would be involved in the disposal process, from what I can tell.

Also forgot to mention this was on dev branch, my mistake there.
